### ISSUE TITLE
Code Completion doesn't work if type the tag name in different case

### ISSUE THREAD

**Issue by atinprominic** [prominic/Moonshine-IDE#158]
I found that the code completion doesn't work if I type the tag name in different case.

For example if I type <js:sim and then press CTRL+Space then it doesn't work. But if I type <js:Sim and then press CTRL+Space then it shows the context menu.

Refer to attached screenshot for details:

![screen shot 2018-01-31 at 4 14 33 pm](https://user-images.githubusercontent.com/34882311/35620523-5f972ec0-06a7-11e8-8c18-6fedbc13ab92.png)

**Comment by piotrzarzycki21** [comment]
@joshtynjala I just check Atin's case and fount that:
1) I'm not getting any items from the language server
2) I'm getting in the console exception:

```
java.lang.ClassCastException: org.apache.royale.compiler.internal.tree.mxml.MXMLFileNode cannot be cast to org.apache.royale.compiler.tree.mxml.IMXMLClassReferenceNode
: 	at com.nextgenactionscript.vscode.ActionScriptTextDocumentService.getEmbeddedActionScriptNodeInMXMLTag(ActionScriptTextDocumentService.java:5486)
: 	at com.nextgenactionscript.vscode.ActionScriptTextDocumentService.hover(ActionScriptTextDocumentService.java:454)
: 	at moonshine.xSocketDataHandler.onData(xSocketDataHandler.java:226)
: 	at org.xsocket.connection.HandlerAdapter.performOnData(HandlerAdapter.java:242)
: 	at org.xsocket.connection.HandlerAdapter.access$200(HandlerAdapter.java:42)
: 	at org.xsocket.connection.HandlerAdapter$PerformOnDataTask.run(HandlerAdapter.java:210)
: 	at org.xsocket.SerializedTaskQueue.performPendingTasks(SerializedTaskQueue.java:161)
: 	at org.xsocket.SerializedTaskQueue.access$100(SerializedTaskQueue.java:40)
: 	at org.xsocket.SerializedTaskQueue$MultithreadedTaskProcessor.run(SerializedTaskQueue.java:189)
: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
: 	at java.lang.Thread.run(Thread.java:748)
: java.lang.ClassCastException: org.apache.royale.compiler.internal.tree.mxml.MXMLFileNode cannot be cast to org.apache.royale.compiler.tree.mxml.IMXMLClassReferenceNode
: 	at com.nextgenactionscript.vscode.ActionScriptTextDocumentService.getEmbeddedActionScriptNodeInMXMLTag(ActionScriptTextDocumentService.java:5486)
: 	at com.nextgenactionscript.vscode.ActionScriptTextDocumentService.hover(ActionScriptTextDocumentService.java:454)
: 	at moonshine.xSocketDataHandler.onData(xSocketDataHandler.java:226)
: 	at org.xsocket.connection.HandlerAdapter.performOnData(HandlerAdapter.java:242)
: 	at org.xsocket.connection.HandlerAdapter.access$200(HandlerAdapter.java:42)
: 	at org.xsocket.connection.HandlerAdapter$PerformOnDataTask.run(HandlerAdapter.java:210)
: 	at org.xsocket.SerializedTaskQueue.performPendingTasks(SerializedTaskQueue.java:161)
: 	at org.xsocket.SerializedTaskQueue.access$100(SerializedTaskQueue.java:40)
: 	at org.xsocket.SerializedTaskQueue$MultithreadedTaskProcessor.run(SerializedTaskQueue.java:189)
: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
: 	at java.lang.Thread.run(Thread.java:748)
```

You have to wait a bit before it's occurs. Initially nothing happen it blows up after couple of seconds. Just hit couple of times CTRL + SPACE (Win). Do you see issue on language server sight ?

**Comment by joshtynjala** [comment]
> I'm not getting any items from the language server

I confirm that pressing Ctrl+Space at the end of `<js:Sim` provides completion items, but not at the end of `<js:sim`. 

However, in VSCode, if I start by typing `<js:` and then continue typing `sim`, it will provide completion items. In this case, VSCode is filtering the list with case-insensitive comparison after the completion items are returned by the language server. In the case that fails, the language server is comparing before returning the completion items, but it seems to be doing a case-sensitive comparison.

> I'm getting in the console exception:

I cannot reproduce this exception. If you can find a way to reproduce, please open a separate issue because it is unrelated.

**Comment by joshtynjala** [comment]
Closed in 81f2edd0cd3aa4f46fd52acc73e7a20c8e19b950

**Comment by JoelProminic** [comment]
I confirmed that the completion works now with exact case match, all lowercase, and all uppercase.
